addons/mail/static/src/core/common/message_reactions.js

```diff
@@ -74,4 +43,26 @@ export class MessageReactions extends Component {
             this.props.openReactionMenu();
         }
     }
+    openReactions(params) {
+        const target = params.ev.currentTarget;
+        const reaction = params.reaction;
+        this.reactionOpened = browser.setTimeout(() => {
+            if (
+                !this.reactionPopover.isOpen ||
+                (this.lastOpenedId && reaction.messageId !== this.lastOpenedId)
+            ) {
+                this.reactionPopover.open(target, {
+                    id: reaction.message.id,
+                    reaction: params.reaction,
+                    openReactionMenu: this.props.openReactionMenu,
+                });
+                this.lastOpenId = reaction.messageId;
+            }
+        }, 350);
+    }
+
+    clearTimeout() {
+        browser.clearTimeout(this.reactionOpened);
+        delete this.reactionOpened;
+    }
```

Comment:
This should be internal code of mouse-hovering to open popover.
Pass options to `usePopover()` so that it knows it's in "hover" mode. This mode put delays on `open/close` and it automatically closes the popover when no longer mouse-overing.

Business code should not have to deal with `timeouts`. This must be all internal code of popover when in this hover mode.

Reply:
Move the timeouts to the popover.